config/config_source.go

```diff
@@ -134,7 +134,12 @@ func NewFileOrEnvironmentSource(
 		config:     cfg,
 	}
 	if configFile != "" {
-		src.watcher.Watch(ctx, []string{configFile})
+		if cfg.Options.IsRuntimeFlagSet(RuntimeFlagConfigHotReload) {
+			src.watcher.Watch(ctx, []string{configFile})
+		} else {
+			log.Warn(ctx).Msg("hot reload disabled")
```

Comment:
I think we say not to log at warning level (see https://www.pomerium.com/docs/community/contributing#logging) — although I'm sure we're not 100% consistent about this. Still, maybe log at `Info()` here?